src/unitxt/task.py

```diff
@@ -38,20 +39,49 @@ class Task(InstanceOperator):
         "metrics" -- to contain the value of Arg 'metrics'
     """
 
-    input_fields: Union[Dict[str, str], List[str]]
-    reference_fields: Union[Dict[str, str], List[str]]
+    input_fields: Optional[Union[Dict[str, str], List[str]]] = None
+    reference_fields: Optional[Union[Dict[str, str], List[str]]] = None
+    inputs: Union[Dict[str, str], List[str]] = DeprecatedField(
+        default=None,
+        metadata={
+            "deprecation_msg": "The 'inputs' field is deprecated. Please use 'input_fields' instead."
+        },
+    )
+    outputs: Union[Dict[str, str], List[str]] = DeprecatedField(
+        default=None,
+        metadata={
+            "deprecation_msg": "The 'outputs' field is deprecated. Please use 'reference_fields' instead."
+        },
+    )
     metrics: List[str]
     prediction_type: Optional[str] = None
     augmentable_inputs: List[str] = []
     defaults: Optional[Dict[str, Any]] = None
 
+    def prepare(self):
+        super().prepare()
+        if (self.input_fields is not None and self.reference_fields is not None) and (
+            self.inputs is not None and self.outputs
+        ):
+            raise ValueError(
+                "Conflicting attributes: 'input_fields' and 'reference_fields' cannot be set simultaneously with 'inputs' and 'outputs'."
+            )
+
+        self.input_fields = (
+            self.input_fields if self.input_fields is not None else self.inputs
+        )
+        self.reference_fields = (
+            self.reference_fields if self.reference_fields is not None else self.outputs
+        )
+
     def verify(self):
         for io_type in ["input_fields", "reference_fields"]:
```

Comment:
We should check that input_fields and reference_fields are not None (because they are optional fields).
Best to check each on separately:

 if (self.input_fields is   None):
            raise ValueError(
                "Missing attribute in task attributes 'input_fields' not set."
            )
 if (self.reference_fields is   None):
            raise ValueError(
                "Missing attribute: 'reference_fields' not set."
            )